presto-hive-common/src/main/java/com/facebook/presto/hive/HiveFileContext.java

```diff
@@ -20,17 +20,20 @@
 
 public class HiveFileContext
 {
-    public static final HiveFileContext DEFAULT_HIVE_FILE_CONTEXT = new HiveFileContext(true, NO_CACHE_CONSTRAINTS, Optional.empty());
+    public static final long UNKNOWN_LENGTH = -1;
+    public static final HiveFileContext DEFAULT_HIVE_FILE_CONTEXT = new HiveFileContext(true, NO_CACHE_CONSTRAINTS, Optional.empty(), UNKNOWN_LENGTH);
 
     private final boolean cacheable;
     private final CacheQuota cacheQuota;
+    private final long fileSize;
```

Comment:
- Move `fileSize` after `extraFileInfo`
- Make the type `OptionalLong` rather than -1 to denote unknown